protocol/builder/response_header_builder.go

```diff
@@ -0,0 +1,60 @@
+package builder
+
+import (
+	"github.com/codecrafters-io/kafka-tester/protocol/api/headers"
+)
+
+type ResponseHeaderBuilder struct {
+	correlationId int32
+	version       int
+}
+
+func NewResponseHeaderBuilder() *ResponseHeaderBuilder {
+	return &ResponseHeaderBuilder{
+		version:       1,
+		correlationId: -1,
+	}
+}
+
+func (rb *ResponseHeaderBuilder) WithCorrelationId(correlationId int32) *ResponseHeaderBuilder {
+	rb.correlationId = correlationId
+	return rb
+}
+
+func (rb *ResponseHeaderBuilder) WithVersion(version int) *ResponseHeaderBuilder {
+	rb.version = version
+	return rb
+}
+
+func (rb *ResponseHeaderBuilder) Build() headers.ResponseHeader {
+	if rb.correlationId == -1 {
+		panic("CodeCrafters Internal Error: Correlation ID is required")
+	}
+
+	if rb.version != 0 && rb.version != 1 {
+		panic("CodeCrafters Internal Error: Version has to be >= 0 and <= 1")
```

Comment:
This is probably better done within `WithVersion` since the default is valid, and that's the only entrypoint to change values. The check above for correlation ID is kinda different, it has to be placed here to see if a user hasn't called `WithCorrelationId